pkgs/by-name/ma/mas/package.nix

```diff
@@ -1,28 +1,27 @@
-{
-  lib,
-  stdenvNoCC,
-  fetchurl,
-  installShellFiles,
-  testers,
-  mas,
-}:
+{ lib, stdenvNoCC, fetchurl, libarchive, p7zip, testers, mas, }:
 
 stdenvNoCC.mkDerivation rec {
   pname = "mas";
-  version = "1.8.6";
+  version = "1.9.0";
 
   src = fetchurl {
-    # Use the tarball until https://github.com/mas-cli/mas/issues/452 is fixed.
-    # Even though it looks like an OS/arch specific build it is actually a universal binary.
-    url = "https://github.com/mas-cli/mas/releases/download/v${version}/mas-${version}.monterey.bottle.tar.gz";
-    sha256 = "0q4skdhymgn5xrwafyisfshx327faia682yv83mf68r61m2jl10d";
+    url =
+      "https://github.com/mas-cli/mas/releases/download/v${version}/mas-${version}.pkg";
+    hash = "sha256-MiSrCHLby3diTAzDPCYX1ZwdmzcHwOx/UJuWrlRJe54=";
   };
 
-  nativeBuildInputs = [ installShellFiles ];
+  nativeBuildInputs = [ libarchive p7zip ];
+
+  unpackPhase = ''
+    7z x $src
+    bsdtar -xf Payload~
+  '';
+
+  dontBuild = true;
 
   installPhase = ''
-    install -D './${version}/bin/mas' "$out/bin/mas"
-    installShellCompletion --cmd mas --bash './${version}/etc/bash_completion.d/mas'
+    mkdir -p $out/bin
+    cp mas $out/bin
   '';
```

Comment:
Preferably add `runHook preInstall` to the start and `runHook postInstall` to the end, but not blocking as it was already like this.

However, this is losing the shell completion. Hopefully it’s present in the `.pkg` and we can install it (and the Fish one too?); see [Homebrew’s formula](https://github.com/Homebrew/homebrew-core/blob/78e62c75bf9f222f69a5a5628b6d09efed499e71/Formula/m/mas.rb#L26-L27). Otherwise, it might be best to stick with the Homebrew bottle for now.

We should be able to build this from source and I might take a look at doing so another time.

Reply:
Unfortunately, no. As I mention in the description, the pkg doesn’t contain any completion files. It looks like the project has never included the completions in the .pkg files, and the tarballs aren’t produced for newer versions. So it looks like we’d wait until this is built from source.